tests/fs_symlink_dir.rs

```diff
@@ -0,0 +1,27 @@
+use std::fs;
+use std::os::unix::fs as unixfs;
+use std::path::PathBuf;
+
+use birdcage::{Birdcage, Exception, Sandbox};
+use tempfile::TempDir;
+
+fn main() {
+    const FILE_CONTENT: &str = "expected content";
+
+    // Setup our test directory.
+    let tempdir = TempDir::new().unwrap();
+    let symlink_str = tempdir.path().to_string_lossy() + "_tmpfile";
+    let symlink_path = PathBuf::from(symlink_str.as_ref());
+    unixfs::symlink(&tempdir, &symlink_path).unwrap();
```

Comment:
There might be some value in adding a broken symlink to one of these tests just to make sure that `birdcage.lock()` still succeeds. Although the symlink should remain broken inside the sandbox, so maybe that is pointless.